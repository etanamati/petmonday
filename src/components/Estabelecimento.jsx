import React from 'react';
import {Card, Button, ListGroup, ListGroupItem} from 'react-bootstrap';
import PropTypes from 'prop-types';

const Estabelecimento = props => {
  const {estabelecimento, handleClick} = props;
  return (
    <Card style={{ width: '20rem', marginTop: '50px' }}>
          <Card.Img variant="top" src={estabelecimento.urlFoto} />
          <Card.Body>
            <Card.Title>{estabelecimento.nome}</Card.Title>
            <Card.Text>
              {estabelecimento.endereco} - {estabelecimento.numero}
            </Card.Text>
          </Card.Body>
          <ListGroup className="list-group-flush">
            <ListGroupItem>Telefone: {estabelecimento.telefone}</ListGroupItem>
            <ListGroupItem>Servi√ßo: {estabelecimento.servico}</ListGroupItem>
          </ListGroup>
          <Card.Body>
            <Button variant="primary" onClick={() => handleClick(estabelecimento.uid)}>Selecione</Button>
          </Card.Body>
        </Card>
  );
}

Estabelecimento.propTypes = {
  estabelecimento: PropTypes.object,
  handleClick: PropTypes.func
}

export default Estabelecimento;
