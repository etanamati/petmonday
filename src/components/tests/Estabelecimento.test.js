import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Estabelecimento from '../Estabelecimento';

it('should render app', () => {
  const estabelecimento = {nome: 'PetShop',
                            numero: '14',
                            telefone: '(44) 3030-7070',
                            endereco: 'Rua Teste',
                            servico: 'Banho e Tosa',
                          urlFoto: ''}
  const wrapper = shallow(<Estabelecimento estabelecimento={estabelecimento}/>);
    
  expect(toJson(wrapper)).toMatchSnapshot();
});