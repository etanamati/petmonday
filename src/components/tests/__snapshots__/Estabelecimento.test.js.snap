// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app 1`] = `
<ForwardRef(Bootstrap(Card))
  style={
    Object {
      "marginTop": "50px",
      "width": "20rem",
    }
  }
>
  <ForwardRef(Bootstrap(CardImg))
    src=""
    variant="top"
  />
  <ForwardRef(Bootstrap(CardBody))>
    <ForwardRef(Bootstrap(CardTitle))>
      PetShop
    </ForwardRef(Bootstrap(CardTitle))>
    <ForwardRef(Bootstrap(CardText))>
      Rua Teste
       - 
      14
    </ForwardRef(Bootstrap(CardText))>
  </ForwardRef(Bootstrap(CardBody))>
  <ForwardRef
    className="list-group-flush"
  >
    <ForwardRef(Bootstrap(ListGroupItem))>
      Telefone: 
      (44) 3030-7070
    </ForwardRef(Bootstrap(ListGroupItem))>
    <ForwardRef(Bootstrap(ListGroupItem))>
      Servi√ßo: 
      Banho e Tosa
    </ForwardRef(Bootstrap(ListGroupItem))>
  </ForwardRef>
  <ForwardRef(Bootstrap(CardBody))>
    <ForwardRef(Bootstrap(Button))
      onClick={[Function]}
      variant="primary"
    >
      Selecione
    </ForwardRef(Bootstrap(Button))>
  </ForwardRef(Bootstrap(CardBody))>
</ForwardRef(Bootstrap(Card))>
`;
